packages/client/src/firebase/firebase.js

```diff
@@ -0,0 +1,16 @@
+import { initializeApp } from 'firebase/app';
+import { getAuth } from 'firebase/auth';
+
+const firebaseConfig = {
+  apiKey: 'AIzaSyARcKPPTVo1aqOCnH-_QAf8rnt3LyUyArY',
```

Comment:
this comment is a complete nit and an FYI (i.e, no changes required), normally, one shouldn't commit keys to the source control. here is a [reading ](https://blog.devgenius.io/quick-tutorial-how-to-deal-with-secret-keys-in-react-apps-97e886c5d2a3)on how you could also do it :)

Reply:
Good point! I have a small question though - we have two env files in the project (one in the server folder, another one in the client folder). The one in the server folder has FIREBASE_APP_API_KEY var to be filled in with the corresponding key. Should we put the key in the env in the server folder then, although the firebase files are in the client folder?

Reply:
If you have ensured that all of these "secrets" can be published (which is what happens with everything that's in the client source, as it gets built and shipped to the user), the key should go into to `client/.env` file, because it's being used in the client. You should prefix it with `REACT_APP_`, so it actually gets included in the build. I haven't read firebase's documentation, so I'm not sure if these keys need to be kept secret or if they can be included in the client build? If you end up adding an environment variable, make sure to ping me, so I can add it to Heroku.

Reply:
I have added an env variable in the client source, works fine. I have also check previous classes' projects - some of them did hide the API key, so let's also stick with it. 

@te-online please add `REACT_APP_FIREBASE_API_KEY=AIzaSyARcKPPTVo1aqOCnH-_QAf8rnt3LyUyArY` to Heroku

Reply:
Okay, I will add it. 

Just to be clear and to add to my last comment, there are 2 different kinds of "secrets" or let's better call them "environment variables" 🤓 

1. Environment variables that can be published (those are not secrets)
2. Environment variables that need to be kept secret (those we can call secrets)

Now, as soon as you use a variable on the client-side, i.e. in a React application it needs to be part of category 1, public, not a secret. That's because it will be exposed in the JS that gets sent to the user's browser.

Because it's sometimes hard to tell if an environment variable is of category 1 or 2, if it's a secret or not, the convention is to never push secrets to git (and ideally to not put them in public comments or similar 😅). Even sending them via Slack is not acceptable. Instead, your organization should have a password manager or similar to share secrets. (I don't think we have that here, so we have to work with what we can get 🤷 )

If you've accidentally published a secret, you _need_ to go back and re-create it. Some will even "purge" the git history of the secret, although that's not strictly needed if it's been re-generated or replaced and the published secret is no longer valid for anything.

I'm assuming this variable is not a secret, because you've put it in the client-side code and you've probably found that in the Firebase docs described like that?

So don't panic when you read this 😅 But keep in mind to always keep any environment variables secret to avoid confusion or accidental publishing of secrets.